examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,241 @@
+#! /bin/python
+"""
+This script will free nodes from a project after
+a specific amount of time.
+Script reads a config file from /etc/leasing.cfg
+which defines which nodes are in the project, time threshold
+admin user name and password and status file.
+
+status file includes each nodes status. which node is in
+the project, which one is free and for how long the node
+has been in the project.
+"""
+
+from sets import Set
+import time
+import ConfigParser
+from os import remove
+from shutil import move
+
+from hil.client.client import Client, RequestsHTTPClient
+from hil.client.base import FailedAPICallException
+
+
+class HILClientFailure(Exception):
+    """Exception indicating that the HIL client failed"""
+
+
+def hil_client_connect(endpoint_ip, name, pw):
+    """Connect to the HIL server and return a HIL Client instance"""
+
+    hil_http_client = RequestsHTTPClient()
+    if not hil_http_client:
+        print('Unable to create HIL HTTP Client')
+        return None
+
+    hil_http_client.auth = (name, pw)
+
+    return Client(endpoint_ip, hil_http_client)
+
+
+def release_nodes(
+        statusfile, non_persistent_list,
+        threshold_time, hil_username, hil_password, hil_endpoint,
+        hil_client=None
+        ):
+
+    """Release nodes : Takes node list as arguments
+    and puts them back to free pool after given time"""
+
+    if not hil_client:
+        hil_client = hil_client_connect(
+                        hil_endpoint, hil_username, hil_password)
+
+    free_node_list = hil_client.node.list('free')
+    # Only these nodes should be updated in
+    # the file(either for project or for time)
+    nodes_to_update_infile = list(Set(
+        non_persistent_list)-Set(free_node_list))
+
+    for node in nodes_to_update_infile:
+        # get the information from status file for node.
+        node_info_from_file = get_project_and_time_for_node(statusfile, node)
+        project_in_file = node_info_from_file[0]
+        old_time = node_info_from_file[1]
+
+        # get the information from HIL for node
+        node_current_info = hil_client.node.show(node)
+        project_in_hil = node_current_info['project']
+        # Increase the time by one unit
+        new_time = int(node_info_from_file[1])+1
+        # See if the node is in the same project and if it is outside
+        # free pool for time less than threshold value then update only
+        # time in status file
+        if (project_in_file == project_in_hil and
+                new_time < threshold_time):
+            update_time_for_node(statusfile, node, old_time, new_time)
+        # See if the node is in the same project and if it is outside
+        # free pool for time greater than threshold value then release
+        # the node back to free pool
```

Comment:
I stopped attempting to read these comments part way through, because they scan poorly (biggest problem is lack of punctuation) and the code is clearer than they are anyway. I don't think they're necessary -- they're basically just re-stating the code in English. In-line comments should provide context, explain *why*, not how -- code is better at the how.